presto-orc/src/main/java/com/facebook/presto/orc/OrcFeature.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc;
+
+/**
+ * Defines features that can be enabled in OrcReader or OrcWriter. A feature
+ * can have an optional config value represented by the generic type of the feature.
+ *
+ * @param <T> the type of this feature's config class
+ */
+public final class OrcFeature<T>
+{
+    public static final OrcFeature<Void> ZSTD_JNI_DECOMPRESSION_ENABLED = new OrcFeature<>();
```

Comment:
The ORCFeature class looks sophisticated. But My worry is generally the features are stored as property bags (string/string value). 

Now all the callers has to parse the string key value in setting up the FeatureFlags correctly.

Usually when a feature has enabled/disabled and optional value, they model it as 2 properties feature.enabled and feature.value.